src/main/java/by/itech/lab/supplier/controller/UserController.java

```diff
@@ -49,37 +55,49 @@
     }
 
     @GetMapping
-    public Page<UserDto> getAllUsers(@PageableDefault Pageable pageable) {
-        return userService.findAll(pageable);
+    public Page<UserDto> getAllByActive(
+            @PageableDefault(sort = {"id"}, direction = Sort.Direction.DESC) final Pageable pageable,
+            @RequestParam(required = false) final Boolean status) {
+        return userService.findAllByActive(pageable, status);
     }
 
-    @GetMapping(ApiConstants.URL_FILTERED)
-    public Page<UserDto> getAllEnabledUsers(@PageableDefault Pageable pageable) {
-        return userService.getAllActive(pageable);
-    }
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
     @AdminAccess
     public UserDto createUser(@Valid @RequestBody UserDto userDto) {
+       CustomerDto customerDto = customerService.findById(userDto.getCustomerId());
+       userDto.setCustomerDto(customerDto);
         return userService.save(userDto);
     }
 
+
+    @GetMapping(ApiConstants.URL_USERNAME_PARAMETER)
+    public Optional<UserDto> getUseByUsername(@PathVariable String username) {
+        log.debug("request to get User : {}", username);
+        return userService.findByUsername(username);
+    }
+
     @ResponseStatus(HttpStatus.OK)
-    @PutMapping(ApiConstants.URL_ID_PARAMETER + ApiConstants.URL_STATUS_PARAMETER)
-    @AdminAccess
-    public int changeActiveStatus(@PathVariable Long id, @PathVariable boolean status) {
-        return userService.changeActiveStatus(id, status);
+    @PutMapping(ApiConstants.URL_ID_PARAMETER + ApiConstants.URL_PASSWORD_PARAMETER)
+    public int changePassword(@PathVariable Long id, @RequestBody String password) {
```

Comment:
please correct me if I wrong: anyone who sent the request to that endpoint is able to change the password for any user? is it expected behaviour?